components/centraldashboard/public/components/activity-view.js

```diff
@@ -1,112 +1,109 @@
-                loading="{{loading}}" on-response="_onResponse">
+            <iron-ajax id="ajax" url="/api/activities/[[namespace]]"
+                handle-as="json" loading="{{loading}}"
+                on-response="_onResponse">
             </iron-ajax>
             <paper-progress indeterminate class="slow"
                 hidden$="[[!loading]]"></paper-progress>
-            <div id="list" role="listbox">
-                <template is="dom-repeat" items="[[activities]]">
-                    <div class="activity-row layout horizontal">
-                        <div class="flex">[[item.formattedTime]]</div>
-                        <div class="flex flex-auto event">
-                            <iron-icon class$="event-icon [[item.icon]]"
-                                icon="[[item.icon]]">
-                            </iron-icon>[[item.event]]
-                        </div>
-                        <div class="flex">[[item.source]]</div>
-                    </div>
-                </template>
-            </div>
+            <p hidden$="[[!message]]">[[message]]</p>
+            <activities-list activities="[[activities]]"></activities-list>
             `;
     }
 
     /**
-      * Object describing property-related metadata used by Polymer features
-      */
+     * Object describing property-related metadata used by Polymer features
+     */
     static get properties() {
         return {
-            loading: Boolean,
-            activities: Array,
+            namespace: {
+                type: String,
+                observer: '_namespaceChanged',
+                value: null,
+            },
+            message: {
+                type: String,
+                value: 'Select a namespace to see recent events',
+            },
+            loading: {
+                type: Boolean,
+                value: false,
+            },
+            activities: {
+                type: Array,
+                value: [],
+            },
         };
     }
 
+    /**
+     * Retrieves Events when namespace is selected.
+     * @param {string} newNamespace
+     */
+    _namespaceChanged(newNamespace) {
+        if (newNamespace) {
+            this.message = '';
+            this.$['ajax'].generateRequest();
+        }
+    }
+
     /**
      * Handles the Activities response to set date format and icon.
      * @param {Event} responseEvent
      */
     _onResponse(responseEvent) {
         const {status, response} = responseEvent.detail;
-        this.activities = [];
-        // TODO: Surface the error in some manner
-        if (status !== 200) return;
-        this.activities = response.map((a) => {
-            const activity = {
-                formattedTime: new Date(a.time).toLocaleString(),
-                icon: a.isError ? 'error' : 'build',
-            };
-            return Object.assign(activity, a);
-        });
+        this.splice('activities', 0);
```

Comment:
I'm sure this has a valid reason, but why do we do this?

Reply:
This makes sure that the the activities list is cleared and then we push the new events onto it later in the method. this.splice and this.push ensure that the changes are notified to other Polymer elements watching the mutable array.